sphinx/application.py

```diff
@@ -968,6 +968,29 @@ def add_transform(self, transform):
         Add the standard docutils :class:`Transform` subclass *transform* to
         the list of transforms that are applied after Sphinx parses a reST
         document.
+
+        .. list-table:: priority range categories for Sphinx transforms
+
+           * - Priority
+             - Main purpose in Sphinx
+           * - 0-99
+             - Fix invalid nodes by docutils. Translate a doctree.
+           * - 100-299
+             - Preparation
+           * - 300-399
+             - early
+           * - 400-699
+             - main
+           * - 700-799
+             - Post processing. Deadline to modify text and referencing.
+           * - 800-899
+             - Collect referencing and referenced nodes. Domain processing.
```

Comment:
Note: At present, domain processing is invoked after transform processing.
With this proposal, I'll move it to here.

Reply:
`doctree-event` is also.

Reply:
I made #4853 for this.